src/scroll-timeline-css-parser.js

```diff
@@ -0,0 +1,381 @@
+    }
+
+    this.assertString(p, "}");
+    const endIndex = p.index;
+    this.eatWhitespace(p);
+    return {
+      scrollTimeline,
+      startIndex,
+      endIndex,
+    };
+  }
+
+  handleScrollTimelineProps(rule, p) {
+    // The animation-timeline property may not be used in keyframes
+    if (rule.selector.includes("@keyframes")) {
+      return;
+    }
+
+    // TODO is it enough to check with "includes()"
+    const hasAnimationName = rule.block.contents.includes("animation-name:");
+    const hasAnimationTimeline = rule.block.contents.includes("animation-timeline:");
+    const hasAnimation = rule.block.contents.includes("animation:");
+
+    let timelineNames = [];
+    let animationNames = [];
+
+    if (hasAnimationTimeline) {
+      timelineNames = this.extract(rule.block.contents, RegexMatcher.ANIMATION_TIMELINE)
+        .split(",").map(name => name.trim());
+    }
+
+    if (hasAnimationName) {
+      animationNames = this.extract(rule.block.contents, RegexMatcher.ANIMATION_NAME)
+        .split(",").map(name => name.trim());
+    }
+
+    if (hasAnimationTimeline && hasAnimationName) {
+      this.saveRelationInList(rule, timelineNames, animationNames);
+      return;
+    }
+
+    if (hasAnimation) {
+      this.extract(rule.block.contents, RegexMatcher.ANIMATION)
+        .split(",").map(shorthand => shorthand.trim()).forEach(shorthand => {
+          const animationName = this.extractAnimationName(shorthand);
+          const timelineName = this.extractTimelineName(shorthand);
+          if (animationName) animationNames.push(animationName);
+          if (timelineName) {
+            timelineNames.push(timelineName);
+            // Remove timeline name from animation shorthand
+            // so the native implementation works with the rest of the properties
+            rule.block.contents = rule.block.contents.replace(
+              timelineName,
+              ""
+            );
+            this.replacePart(
+              rule.block.startIndex,
+              rule.block.endIndex,
+              rule.block.contents,
+              p
+            );
+          }
+        });
+    }
+
+    this.saveRelationInList(rule, timelineNames, animationNames);
+  }
+
+  saveRelationInList(rule, timelineNames, animationNames) {
+    if (animationNames.length == 0) {
+      for (let i = 0; i < timelineNames.length; i++) {
+        this.cssRulesWithTimelineName.push({
+          selector: rule.selector,
+          'animation-name': undefined,
+          'animation-timeline': timelineNames[i]
+        });
+      }
+    } else {
+      for (let i = 0; i < Math.max(timelineNames.length, animationNames.length); i++) {
+        this.cssRulesWithTimelineName.push({
+          selector: rule.selector,
+          'animation-name': animationNames[i % animationNames.length],
+          'animation-timeline': timelineNames[i % timelineNames.length]
+        });
+      }
+    }
+
+  }
+
+  extractAnimationName(shorthand) {
```

Comment:
This and the following extract functions are nearly identical. They can call a common method that checks if a name is in a map. e.g.

extractAnimationName(shorthand) {
  return this, findMatchingEntryInMap(shorthand, this.keyframeNames);
}
findMatchingEntryInMap(shorthand, map) {
   const matches = shorthand.split(" ").filter(part => map.has(part));
   return matches ? matches[0] : null;
}